### PR TITLE
[workspace] Converted to Yarn Workspace 3 with Lerna

### PR DESCRIPTION
**Pull request by nebula-aac**
**Notes for Reviewers**

This PR fixes https://github.com/layer5io/sistent/issues/5

I have another branch for converting to Yarn 3 Workspace.

Since we're already using Yarn, we can definitely turn this into a true workspace, and use the latest from Yarn 3. This is an optional branch for anyone to review.

Breaking Changes
- Design System has been moved to `apps`, make sense to move it here, instead of being involved in the build process. This project has also been migrated to Vite Typescript.
- `svg` and `components` have officially being migrated to use Vite.
- Two new release workflows to build two of the packages separately
- Fixed where we can run `lint` from workspace, which was missing out the linting from each of the packages

**UPDATED**
- Updated forked to include the `ErrorBoundary` in `components`
- Updated forked to include `rtk-query-codegen`
- Updated `ErrorBoundary` to define custom type definitions
- Updated `rtk-query-codegen` with scripts in place to build

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@Mohith234 you might like to review (even after merge)

**Comment by leecalcote** [comment]
> This project has also been migrated to Vite Typescript.

✔️

File: .github/workflows/components-release.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Build and Publish Components Package
+
+on:
+  release:
+    types: [published]    
+  workflow_dispatch:
+    inputs:
```
Comment (leecalcote) [inline .github/workflows/components-release.yml:7]: While dispatching, if one of the steps here isn't dynamically identifying the latest release tag, we might want to require the release tag as input here. 

You might like to grab the release tag lookup "function" from another one of the makefiles / workflows.

File: .github/workflows/components-release.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  publish-gpr:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
```
Comment (leecalcote) [inline .github/workflows/components-release.yml:40]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/components-release.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
```
Comment (leecalcote) [inline .github/workflows/components-release.yml:19]:
```suggestion
        uses: actions/checkout@v4
```

**Comment by saurabh100ni** [comment]
Let's discuss these updates on the website call today
Please add this as an agenda item in the meeting minutes, if you would :)
https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima

**Pull request merged**